pymc/distributions/logprob.py

```diff
@@ -244,12 +244,12 @@ def joint_logpt(
             and not isinstance(node.owner.op, SimulatorRV)
         )
     ]
-    if unexpected_rv_nodes:
-        raise ValueError(
-            f"Random variables detected in the logp graph: {unexpected_rv_nodes}.\n"
-            "This can happen when DensityDist logp or Interval transform functions "
-            "reference nonlocal variables."
-        )
```

Comment:
Nodes appear because, when `size` is not specified, `init_dist` gets broadcasted with respected to `mu` and `sigma` and, as such, they will appear in the `logp` graph. The presence of these random variables raises these errors.

Reply:
The shape can be evaluated akin to what is done in the `AR` distribution below

https://github.com/pymc-devs/pymc/blob/729f79caa68551a9c86c47ac14e623518b099835/pymc/distributions/timeseries.py#L493-L499